python/lsst/daf/persistence/butler.py

```diff
@@ -1245,17 +1245,23 @@ def datasetExists(self, datasetType, dataId={}, **rest):
         datasetType = self._resolveDatasetTypeAlias(datasetType)
         dataId = DataId(dataId)
         dataId.update(**rest)
+        location = self._locate(datasetType, dataId, write=False)
+        if location is None:
+            return False
 
-        location = None
-        for repoData in self._repos.inputs():
-            if not dataId.tag or len(dataId.tag.intersection(repoData.tags)) > 0:
-                location = repoData.repo.map(datasetType, dataId)
-                if location and location.repository.exists(location):
-                    break
+        exists = None
+        if isinstance(location, ButlerComposite):
+            for name, componentInfo in location.componentInfo.items():
+                if componentInfo.subset:
+                    subset = self.subset(datasetType=componentInfo.datasetType, dataId=location.dataId)
+                    exists = all([obj.datasetExists() for obj in subset])
                 else:
-                    location = None
-
-        return bool(location)
+                    exists = self.datasetExists(componentInfo.datasetType, location.dataId)
```

Comment:
The indentation looks odd.  I believe that we are sticking to the 110 line limit; how long is this one?